src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java

```diff
@@ -15,191 +11,232 @@
+        Map<String, Object> response = new HashMap<>();
+        logger.info(String.format("Updating User Login: rootOrg: %s, userId: %s", userLoginInfo.getOrgId(),
+                userLoginInfo.getUserId()));
+        userLoginInfo.setLoginTime(new Timestamp(new Date().getTime()));
+        try {
+            Map<String, Object> propertyMap = new HashMap<>();
+            propertyMap.put(Constants.USER_ID, userLoginInfo.getUserId());
+            List<Map<String, Object>> details = cassandraOperation.getRecordsByProperties(Constants.DATABASE,
+                    Constants.LOGIN_TABLE, propertyMap, null);
+            if (CollectionUtils.isEmpty(details)) {
+                Map<String, Object> request = new HashMap<>();
+                request.put(Constants.USER_ID, userLoginInfo.getUserId());
+                request.put(Constants.FIRST_LOGIN_TIME, userLoginInfo.getLoginTime());
+                cassandraOperation.insertRecord(Constants.DATABASE, Constants.LOGIN_TABLE, request);
+                telemetryUtils.pushDataToKafka(telemetryUtils.getUserslastLoginTelemetryEventData(userLoginInfo),
+                        serverConfig.getUserUtilityTopic());
+            }
+        } catch (Exception e) {
+            throw new ApplicationLogicError("Sunbird Service ERROR: ", e);
+        }
+        response.put(userLoginInfo.getUserId(), Boolean.TRUE);
+        return response;
+    }
+
+    @Override
+    public Map<String, Object> getUsersReadData(String userId, String authToken, String X_authToken) {
+        Map<String, String> header = new HashMap<>();
+        header.put(Constants.AUTH_TOKEN, authToken);
+        header.put(Constants.X_AUTH_TOKEN, X_authToken);
+        Map<String, Object> readData = (Map<String, Object>) outboundRequestHandlerService
+                .fetchUsingGetWithHeadersProfile(serverConfig.getSbUrl() + serverConfig.getLmsUserReadPath() + userId,
+                        header);
+        Map<String, Object> result = (Map<String, Object>) readData.get(Constants.RESULT);
+        Map<String, Object> responseMap = (Map<String, Object>) result.get(Constants.RESPONSE);
+        return responseMap;
+    }
+
+    @Override
+    public Map<String, Object> autoSearchUser(String key) throws IOException {
+        Map<String, Object> searchQueryMap = new HashMap<>();
+        String field = getInstance().getProperty("user.autocomplete.es.keys");
+        //prepare searchQuerymap
+        final String[] includeFields = field.split(",");
+        Map<String, List<String>> querySearchFields = new HashMap();
+        querySearchFields.put(key, Arrays.asList(includeFields));
+        searchQueryMap.put(Constants.MULTI_QUERY_SEARCH_FIELDS, querySearchFields);
+        SearchDTO searchDto = ElasticSearchHelper.createSearchDTO(searchQueryMap);
+        searchDto.setExcludedFields(Arrays.asList(ProjectUtil.excludes));
+        Map<String, Object> result = esService.search(searchDto, ProjectUtil.EsType.user.getTypeName(), esClient);
```

Comment:
esService should be from autoWired and esClient can be initialised inside esService itself instead of passing it from here